t/chaos/killetcd/killetcd.go

```diff
@@ -96,8 +96,23 @@ var _ = ginkgo.Describe("Test Get Success When Etcd Got Killed", func() {
 	ginkgo.It("check if everything works", func() {
 		utils.SetRoute(e, httpexpect.Status2xx)
 		utils.GetRouteList(e, http.StatusOK)
-		time.Sleep(1 * time.Second)
-		utils.GetRoute(e, http.StatusOK)
+		var resp *httpexpect.Response
+
+		resp = utils.GetRouteIgnoreError(e)
+		// wait 1s seems not enough, wait some more time to make sure nothing goes wrong
+		if resp.Raw().StatusCode != http.StatusOK {
```

Comment:
Try to use `k8s.io/apimachinery/pkg/utils/wait`: https://pkg.go.dev/k8s.io/apimachinery@v0.21.3/pkg/util/wait

Reply:
Actually there are already [checks](https://github.com/apache/apisix/pull/4654/files/39a5af086a7a48d752f7cd8a052459945d8a2fa4#diff-aef74626794d502616587a665ada7300d710401a1ceff2c687e9f30cb4975b6bR66) on apisix deployment when start apisix. But the problem - **new configuration needs seconds to be synced** - still exists.

The possible reason I guess is related to https://github.com/apache/apisix-dashboard/pull/976 and it seems we need longer waiting time for this situation in kubernetes. Have you met with this problem before?

Reply:
Nope.